### ISSUE TITLE
Missing predicate functions for `UnnecessaryFilter`

### ISSUE THREAD

**Issue by marschwar** [detekt/detekt#6434]
## Expected Behavior of the rule
The `UnnecessaryFilter`  rule should flag all terminal operations that have an equivalent that takes a predicate as parameter. 

```kotlin
// non-compliant
val x = listOf(1, 2, 3)
     .filter { it > 1 }
     .first()

// compliant
val x = listOf(1, 2, 3)
     .first { it > 1 }
```

The list should include at least: `any()`, `none()`,  `first()`, `firstOrNull()`, `last()`, `lastOrNull()`, `single()`, `singleOrNull()`.
